### PR TITLE
Add legacyReviewId to reviews

### PR DESCRIPTION
**Pull request by meshde**
* Add `legacyReviewId` to review endpoints
* Add `review.legacyReviewId` filter to `GET /submissions` endpoint

Fix for #143

### CODE REVIEW

**Comment by callmekatootie** [comment]
@cwdcwd This one's good to merge

File: src/services/ReviewService.js [hunk 1]
```diff
@@ -181,6 +182,9 @@ createReview.schema = {
     .object()
     .keys({
       score: joi.score().required(),
+      legacyReviewId: joi
+        .string()
```
Comment (cwdcwd) [inline src/services/ReviewService.js]: This actually is a number in the legacy systems and not a UUID

File: src/services/ReviewService.js [hunk 2]
```diff
@@ -312,6 +317,9 @@ updateReview.schema = {
...
```
Comment (cwdcwd) [inline src/services/ReviewService.js]: same here. number, not a string

File: src/services/SubmissionService.js [hunk 3]
```diff
@@ -205,6 +205,7 @@ const listSubmissionsQuerySchema = {
   perPage: joi.pageSize(),
   orderBy: joi.sortOrder(),
   'review.score': joi.score(),
+  'review.legacyReviewId': joi.string().uuid(),
```
Comment (cwdcwd) [inline src/services/SubmissionService.js]: same here. number, not a string

File: docs/swagger.yaml [hunk 4]
```diff
@@ -928,6 +930,7 @@ paths:
             example:
               - id: 'a12a4180-65aa-42ec-a945-5fd21dec0503'
                 score: 95.5
+                legacyReviewId: '6b4faff1-7685-4866-a2e3-24afabf5c2c1'
```
Comment (cwdcwd) [inline docs/swagger.yaml]: same here. number, not a string

File: docs/swagger.yaml [hunk 5]
```diff
@@ -1610,6 +1615,13 @@ parameters:
     required: false
     type: integer
 
+  filterSubmissionReviewLegacyReviewId:
+    in: query
+    name: review.legacyReviewId
+    description: The legacy review id filter of the reviews associated with the submission.
+    required: false
+    type: string
```
Comment (cwdcwd) [inline docs/swagger.yaml]: same here. number, not a string

File: docs/swagger.yaml [hunk 6]
```diff
@@ -1988,6 +2008,10 @@ definitions:
         format: double
         description: The review score.
         example: 95.5
+      legacyReviewId:
+        type: string
```
Comment (cwdcwd) [inline docs/swagger.yaml]: same here. number, not a string

**Comment by meshde** [comment]
> Please change from a UUID string to a number field

@cwdcwd updated

**Comment by callmekatootie** [comment]
@cwdcwd The merge conflicts are resolved.

**Pull request merged**